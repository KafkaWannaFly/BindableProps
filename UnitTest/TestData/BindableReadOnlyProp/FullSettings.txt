namespace UnitTest.TestData.UnitTest;

public partial class MyClass
{
    // Create prop with a few settings
    [BindableReadOnlyProp(DefaultBindingMode = ((int)BindingMode.TwoWay))]
    string text = "From every time";

    // Full setting
    [BindableReadOnlyProp(
        DefaultBindingMode = ((int)BindingMode.OneWay),
        ValidateValueDelegate = nameof(ValidateValue),
        PropertyChangedDelegate = nameof(PropertyChangedDelegate),
        PropertyChangingDelegate = nameof(PropertyChangingDelegate),
        CoerceValueDelegate = nameof(CoerceValueDelegate),
        CreateDefaultValueDelegate = nameof(CreateDefaultValueDelegate)
        )]
    string placeHolder = "Always!";

    static bool ValidateValue(BindableObject bindable, object value)
    {
        return true;
    }

    static void PropertyChangedDelegate(BindableObject bindable, object oldValue, object newValue)
    {
        // Do something
    }

    static void PropertyChangingDelegate(BindableObject bindable, object oldValue, object newValue)
    {
        // Do something
    }

    static object CoerceValueDelegate(BindableObject bindable, object value)
    {
        // Do something
        return 0;
    }

    static object CreateDefaultValueDelegate(BindableObject bindable)
    {
        // Do something
        return string.Empty;
    }
}