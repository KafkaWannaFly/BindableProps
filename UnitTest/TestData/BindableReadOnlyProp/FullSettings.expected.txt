#pragma warning disable

// <auto-generated/>

namespace UnitTest.TestData.UnitTest
{
    public partial class MyClass
    {
        public static readonly BindablePropertyKey TextPropertyKey = BindableProperty.CreateReadOnly(
                            nameof(Text),
                            typeof(string),
                            typeof(MyClass),
                            "From every time",
                            (BindingMode)((int)BindingMode.TwoWay),
                            null,
                            (bindable, oldValue, newValue) =>
                                        ((MyClass) bindable).Text = (string) newValue,
                            null,
                            null,
                            null
                        );

        public static readonly BindableProperty TextProperty = TextPropertyKey.BindableProperty;

        public string Text
        {
            get => text;
            private set
            {
                OnPropertyChanging(nameof(Text));

                text = value;
                SetValue(MyClass.TextPropertyKey, text);

                OnPropertyChanged(nameof(Text));
            }
        }

        public static readonly BindablePropertyKey PlaceHolderPropertyKey = BindableProperty.CreateReadOnly(
                    nameof(PlaceHolder),
                    typeof(string),
                    typeof(MyClass),
                    "Always!",
                    (BindingMode)((int)BindingMode.OneWay),
                    ValidateValue,
                    PropertyChangedDelegate,
                    PropertyChangingDelegate,
                    CoerceValueDelegate,
                    CreateDefaultValueDelegate
                );

        public static readonly BindableProperty PlaceHolderProperty = PlaceHolderPropertyKey.BindableProperty;

        public string PlaceHolder
        {
            get => placeHolder;
            private set
            {
                OnPropertyChanging(nameof(PlaceHolder));

                placeHolder = value;
                SetValue(MyClass.PlaceHolderPropertyKey, placeHolder);

                OnPropertyChanged(nameof(PlaceHolder));
            }
        }
    }
}